samcli/commands/build/click_container.py

```diff
@@ -0,0 +1,27 @@
+"""
+Module to check container based cli parameters
+"""
+import click
+
+
+class ContainerOptions(click.Option):
+    """
+    Preprocessing checks for presence of --use-container flag for container based options.
+    """
+
+    def __init__(self, *args, **kwargs):
+        self.require_container: bool = kwargs.pop("require_container", False)
+
+        super().__init__(*args, **kwargs)
+
+    def handle_parse_result(self, ctx, opts, args):
+        if self.require_container:
+            if "use_container" not in opts and opts.get(self.name) is not None:
+                msg = f"""\
+Missing required parameter, with --{self.name.replace("_", "-")} set.
```

Comment:
This seems redundant, the next sentence basically says the same thing.

Reply:
this is telling which parameter requires the -u flag, e.g., --build-image

Reply:
I am not sure I understand, when this error message is printed, it looks like this:
```
Missing required parameter, with --build-image set.
Must provide the --use-container flag in order to use --build-image flag.
```

I think the 2nd sentence contains all the information in the first one, what do you think?

Reply:
Sounds fair, I'll combine them